### DISCUSSION TITLE
GNU GPL license abuse of Unicornsoft Co Ltd

### DISCUSSION THREAD

**Discussion by piquark6046** [uBlockOrigin/uBlock-issues#2650] [General]
Unicornsoft Co Ltd, developer of **[Unicorn Pro adblocker](https://getunicorn.app/ko)**, published Unicorn Pro for Android 1.17.143(216).
However, **some parts of uBlock Origin is stolen** by them.

### Screenshots

<details><summary>setInterval-defuser.js</summary>

![Screenshot_from_2023-05-09_23-18-26 cleaned](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/0b0963ea-176a-4f04-a52e-62aabe64da7b)

</details> 

<details><summary>abort-current-inline-script.js</summary>

![Screenshot_from_2023-05-09_23-21-58 cleaned](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/ae7bfb0b-a82e-4c50-b387-402c1e85d826)

</details> 

### References:
 - https://getunicorn.app/ko/release-note/unicorn-pro/android - [Archive](https://web.archive.org/web/20230510141006/https://getunicorn.app/ko/release-note/unicorn-pro/android)

**Comment by piquark6046** [comment]
<details><summary>abort-on-property-write.js</summary>

![Screenshot from 2023-05-10 23-41-33](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/95d46a60-7519-4f09-9ec9-042faccebfb1)

</details> 

<details><summary>addEventListener-defuser.js</summary>

![Screenshot from 2023-05-10 23-51-10](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/7a37211d-89ae-49d4-8a70-7142e34b9d25)


</details> 

<details><summary>Extended CSS :has-text</summary>

![Screenshot from 2023-05-11 00-02-04](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/2a671ad6-605f-4d5f-b787-0559ee7135f2)
![Screenshot from 2023-05-11 00-08-00](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/647f3ae9-1c9a-4bff-8b52-aefe51054214)
![Screenshot from 2023-05-11 00-11-54](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/a425bde2-2a4a-416f-9bc7-88bd9bebcbfe)


</details>

**Comment by piquark6046** [comment]
Update: Unicornsoft Co Ltd added slimier extended CSS and scriptlets into Unicorn Pro for Windows. Will be checked.
https://getunicorn.app/ko/release-note/unicorn-pro/windows - [Archive](https://web.archive.org/web/20230511123654/https://getunicorn.app/ko/release-note/unicorn-pro/windows)

**Reply by piquark6046** [reply]
<details><summary>:has-text</summary>


![Screenshot from 2023-05-14 16-50-18](https://github.com/uBlockOrigin/uBlock-issues/assets/98787049/0d25e0e5-90e4-4f4f-9118-34a8dc7234b1)


</details> 

Same. But, a scriptlet filter written same in Unicorn Pro for Android does not work in Windows version.
It seems that it is an ineffectual bug.

**Comment by piquark6046** [comment]
I think that the reason why Unicornsoft wants to support advanced syntax existing in List-KR filters list by seeing its community responses.
But, I will **NEVER** support it. 

@gorhill How will you handle the issue?

**Reply by gorhill** [reply]
> @gorhill How will you handle the issue?

I don't intend to do anything, this project is a hobby for me, I am not a corporation with a legal department, and I don't want to waste precious personal time on such issue. When I do see such sleazy behavior, I just point out at it publicly and that's it.

**Reply by piquark6046** [reply]
I guess that they have used source code of uBlock Origin because uBlock Origin developer does not waste their personal time on such issue even though Unicorn Pro adblocker has some nicknames such as K-애드가드 [^1] and 유니가드 [^2].
And, Unicornsoft replied that AdGuard version of List-KR filter list is recommended when they receive that a user wants to add List-KR filter into Unicorn Pro adblocker in their official Telegram chat [^3].

[^1]: The `K-` prefix is a short word for Korean or Korea. The Korean-language word `애드가드` means AdGuard.
[^2]: _Uniguard_. A mixed word containing Unicorn and AdGuard.
[^3]: https://t.me/unicorn_app_chat/4923

**Reply by piquark6046** [reply]
Also, Unicorn Pro adblocker has been compared with AdGuard even in [its official forum](https://support.getunicorn.app/hc/ko/community/topics).

**Reply by piquark6046** [reply]
I think that `@gorhill` can take down easily Unicorn Pro adblocker for iOS if they start to include the source code into it.
https://www.apple.com/legal/internet-services/itunes/appstorenotices/#?lang=en
https://www.apple.com/legal/contact/copyright-infringement.html

**Comment by nicekon** [comment]
Hello @gorhill,

My name is Kwon from the Unicorn development team. Recently, we have referenced and modified some of the open-source code distributed by uBlock for our expanded CSS and scriptlets filter functionality.

However, we became aware that the code we used is subject to the GPLv3 license, which obligates us to either disclose all source code or provide a written offer valid for three years to supply the source code upon request.

Upon this realization, we immediately rolled back the code containing this functionality and released a new version with the relevant functions removed. This applied to the following products and versions:

Unicorn Pro (Windows) 1.17.143(216)
Unicorn Pro (Android) 1.5.210.1294

We now plan to independently redesign and support the aforementioned functionality. This experience has led us to approach open-source license policies more carefully. We have shared these facts on the Unicorn forum and notified our users.(https://support.getunicorn.app/hc/ko/community/posts/18816691323929)

While we do not have any code to disclose at the moment, we plan to fulfill our obligations under the GPLv3 license in the future. We wanted to inform you of this matter to prevent any misunderstanding between the Unicorn development team and the uBlock team.

Please feel free to contact us if you or the uBlock team have any questions or need additional information.

Thank you,

Kwon
Unicorn Development Team

**Comment by piquark6046** [comment]
@nicekon 
Can you tell us how your team get the following rules at [the official forum](https://support.getunicorn.app/hc/ko/community/posts/33234486343321/comments/33590130857497)?
```adblock
||www.youtube.com/playlist?list=$xhr,1p,replace=/"adPlacements.*?([A-Z]"\}|"\}{2\,4})\}\]\,//
||www.youtube.com/playlist?list=$xhr,1p,replace=/"adSlots.*?\}\]\}\}\]\,//
```

**Reply by nicekon** [reply]
Hello, piquark6046,

Regarding the issue you raised, we would like to provide an explanation.

The filter rules mentioned in the community were temporarily referenced during the customer support process to resolve a specific issue. These rules were provided as test references to quickly address the customer’s issue and are different from the official filter rules of Unicorn Soft.

Unicorn Soft’s official filter rules are internally developed and optimized. The temporary rules provided in the community were used solely for testing purposes and are managed separately from our official rules.

Additionally, while the CS team provided these rules for reference during customer support, any reported issues are actually resolved by our development team, who write the necessary filters themselves. This is part of our process to ensure prompt and accurate problem resolution for our customers.

Moving forward, we will be more careful to avoid any misunderstandings during customer support interactions.

We apologize for any inconvenience this issue may have caused. If you have any further questions or feedback, please feel free to contact us at any time.

Thank you.
Unicorn Soft Team
